packages/libraries/cli/src/base-command.ts

```diff
@@ -4,16 +4,38 @@ import type { ExecutionResult } from 'graphql';
 import { http } from '@graphql-hive/core';
 import type { TypedDocumentNode } from '@graphql-typed-document-node/core';
 import { Command, Errors, Flags, Interfaces } from '@oclif/core';
+import { CommandError } from '@oclif/core/lib/interfaces';
 import { Config, GetConfigurationValueType, ValidConfigurationKeys } from './helpers/config';
+import { OmitNever } from './helpers/general';
+import { Envelope, OutputType } from './helpers/output-schema';
+import { Typebox } from './helpers/typebox/__';
 
-export type Flags<T extends typeof Command> = Interfaces.InferredFlags<
-  (typeof BaseCommand)['baseFlags'] & T['flags']
->;
-export type Args<T extends typeof Command> = Interfaces.InferredArgs<T['args']>;
+export default abstract class BaseCommand<$Command extends typeof Command> extends Command {
```

Comment:
I often use `$` prefix type names to maintain readability whilst also separating type parameters from other types.

Reply:
I think we are not using this approach in the codebase at the moment, but I don't have any strong opinion here. It seems readable and easy to understand.